File: crates/ecp-cli/tests/fixtures/incremental_parity/php/UserService.php
<?php

namespace App\Services;

use App\Models\User;

class UserService
{
    private UserRepository $repository;

    public function __construct(UserRepository $repository)
    {
        $this->repository = $repository;
    }

    public function findById(int $id): ?User
    {
        return $this->repository->findById($id);
    }

    public function findAll(): array
    {
        return $this->repository->findAll();
    }

    public function create(string $email, string $name): User
    {
        $user = new User(0, $email, $name);
        return $this->repository->save($user);
    }

    public function delete(int $id): bool
    {
        return $this->repository->delete($id);
    }
}

File: crates/ecp-cli/tests/fixtures/incremental_parity/java/UserService.java
package com.example;

import java.util.List;
import java.util.Optional;

public class UserService {
    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public Optional<User> findById(long id) {
        return repository.findById(id);
    }

    public List<User> findAll() {
        return repository.findAll();
    }

    public User save(User user) {
        return repository.save(user);
    }

    public void delete(long id) {
        repository.deleteById(id);
    }
}
